### PR TITLE
Apply dataSource changes only after the widget's radios collection is ready (T861468)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/radio_group/radio_group.js [hunk 1]
```diff
@@ -260,6 +261,7 @@ class RadioGroup extends Editor {
     }
 
     _renderRadios() {
+        this.isRadiosRendered = new Deferred();
```
Comment (San4es) [inline js/ui/radio_group/radio_group.js]:
```suggestion
        this._isRadiosRendered = new Deferred();
```

File: js/ui/radio_group/radio_group.js [hunk 2]
```diff
@@ -278,6 +280,8 @@ class RadioGroup extends Editor {
             selectedItemKeys: [this.option('value')],
             tabIndex: this.option('tabIndex')
         });
+
+        this.isRadiosRendered.resolve();
```
Comment (San4es) [inline js/ui/radio_group/radio_group.js]:
```suggestion
        this._isRadiosRendered.resolve();
```

File: js/ui/radio_group/radio_group.js [hunk 3]
```diff
@@ -302,7 +306,7 @@ class RadioGroup extends Editor {
     }
 
     _setCollectionWidgetOption() {
-        this._setWidgetOption('_radios', arguments);
+        this.isRadiosRendered.done(this._setWidgetOption.bind(this, '_radios', arguments));
```
Comment (San4es) [inline js/ui/radio_group/radio_group.js]:
```suggestion
        this._isRadiosRendered.done(this._setWidgetOption.bind(this, '_radios', arguments));
```

File: js/ui/radio_group/radio_group.js (hunk 2 above)
Comment (San4es) [inline js/ui/radio_group/radio_group.js]: Do we need to move this to `_fireContentReadyAction`? I suggest writing a test with an asynchronous data source to ensure this.

File: testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js [hunk 4]
```diff
@@ -136,6 +136,25 @@ module('buttons group rendering', () => {
         assert.strictEqual(onContentReadyHandler.callCount, 2);
     });
 
+    test('should render new items if them were setted in onContentReady handler (T861468)', function(assert) {
+        const onContentReadyHandler = sinon.spy((e) => {
+            if(!isReady) {
+                isReady = true;
+                e.component.option('items', [1, 2, 3]);
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js:143]: image

See:
https://js.devexpress.com/Documentation/20_1/ApiReference/UI_Widgets/dxRadioGroup/Configuration/items/
https://js.devexpress.com/Documentation/20_1/ApiReference/UI_Widgets/dxRadioGroup/Configuration/#dataSource

File: js/ui/radio_group/radio_group.js (hunk 2 above)
Comment (alexander-kotov-dx) [reply js/ui/radio_group/radio_group.js]: I have added the test

File: js/ui/radio_group/radio_group.js [hunk 5]
```diff
@@ -260,6 +261,7 @@ class RadioGroup extends Editor {
     }
 
     _renderRadios() {
+        this._isRadiosInited = new Deferred();
```
Comment (San4es) [inline js/ui/radio_group/radio_group.js]:
```suggestion
        this._areRadiosCreated = new Deferred();
```

**Review by San4es — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**